src/System.Buffers/ref/System.Buffers.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// ------------------------------------------------------------------------------
+// Changes to this file must follow the http://aka.ms/api-review process.
+// ------------------------------------------------------------------------------
+
+namespace System.Buffers
+{
+    public abstract class ArrayPool<T>
+    {
+        public const int DefaultNumberOfArraysPerBucket = 50;
+        public const int DefaultArrayLength = 1024000;
+
+        public static ArrayPool<T> Shared{ get; }
+        public static ArrayPool<T> Create(int maxArrayLength = DefaultArrayLength, int arraysPerBucket = DefaultNumberOfArraysPerBucket);
+        public abstract int RentedBufferCount { get; }
+        public abstract int TotalBufferCount { get; }
```

Comment:
Ah okay, that would make more sense then. Do you mean hooking up to EventSource and providing more diagnostic data through there? That seems like the RentedCount could be provided through there as well since that seems like diagnostic information too